src/controllers/authController.js

```diff
@@ -0,0 +1,156 @@
+'use strict';
+
+const bcrypt = require('bcrypt');
+const userSevices = require('../services/userServices.js');
+const jwtServices = require('../services/jwtServices.js');
+const { User } = require('../models/user.js');
+const { ApiError } = require('../exeption/apiError.js');
+const tokenSevices = require('../services/tokenServices.js');
+const validate = require('../utils/validate.js');
+
+const register = async(req, res) => {
+  const { name, email, password } = req.body;
+
+  const errors = {
+    email: validate.email(email),
+    password: validate.password(password),
+    name: validate.username(name),
+  };
+
+  if (!errors.email || !errors.password || !errors.name) {
```

Comment:
Consider making validations in the middleware before controllers and sending the 400 error here